base/essentials.jl

```diff
@@ -488,10 +488,24 @@ macro inbounds(blk)
         Expr(:inbounds, :pop))
 end
 
+"""
+    @label name
+
+Labels a statement with the symbolic label `name`. The label marks the end-point
+of an unconditional jump with [`@goto name`](@ref).
+"""
 macro label(name::Symbol)
     return esc(Expr(:symboliclabel, name))
 end
 
+"""
+    @goto name
+
+`@goto name` unconditionally jumps to the statement at the location [`@label name`](@ref).
+
+`@label` and `@goto` cannot create jumps to top-level statements. Attempts cause an
+error. To use `@goto` at the top-level, enclose the `@label` and `@goto` in a block.
```

Comment:
This is kind of a contradiction – "to use `@goto` at the top level, don't use `@goto` at the top level." Or maybe it's a koan.

Reply:
"To still use `@goto` in this situation..." works